operator/pkg/helmreconciler/render.go

```diff
@@ -386,6 +367,71 @@ func toChartManifestsMap(m name.ManifestMap) ChartManifestsMap {
 	return out
 }
 
+// getOwnerLabels returns a map of labels for the given component name, revision and owning CR resource name.
+func (h *HelmReconciler) getOwnerLabels(componentName string) (map[string]string, error) {
+	crName, err := h.getCRName()
+	if err != nil {
+		return nil, err
+	}
+	labels := make(map[string]string)
+	revision := ""
+	if h.iop != nil {
+		revision = h.iop.Spec.Revision
+	}
+
+	componentLabelValue := componentName
+	// Only pilot component uses revisions
+	if componentName == string(name.PilotComponentName) {
+		if revision == "" {
+			revision = "default"
+		}
+		labels[model.RevisionLabel] = revision
+		componentLabelValue += "-" + revision
+	}
+
+	labels[operatorLabelStr] = operatorReconcileStr
+	labels[owningResourceKey] = crName
+	labels[istioComponentLabelStr] = componentLabelValue
+	labels[istioVersionLabelStr] = pkgversion.Info.Version
+
+	return labels, nil
+}
+
+// applyLabelsAndAnnotations applies owner labels and annotations to the object.
+func (h *HelmReconciler) applyLabelsAndAnnotations(obj runtime.Object, componentName string) error {
+	labels, err := h.getOwnerLabels(componentName)
+	if err != nil {
+		return err
+	}
+
+	for k, v := range labels {
+		err := util.SetLabel(obj, k, v)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (h *HelmReconciler) getCRName() (string, error) {
+	if h.iop == nil {
+		return "", nil
+	}
+	objAccessor, err := meta.Accessor(h.iop)
+	if err != nil {
+		return "", err
+	}
+	return objAccessor.GetName(), nil
+}
+
+func (h *HelmReconciler) getCRHash(componentName string) (string, error) {
+	crName, err := h.getCRName()
```

Comment:
name is not unique -  should we have namespace here?

Reply:
yes, good point.

Reply:
done.